yoti_python_sdk/activity_details.py

```diff
@@ -42,33 +48,56 @@ def try_parse_selfie_field(self, field):
             field.content_type
         )
 
-    def try_parse_age_verified_field(self, field):
+    def try_parse_age_verified_field(self, field, anchors):
         if field is not None:
             is_age_verified = Protobuf().value_based_on_content_type(
                 field.value,
                 field.content_type
             )
             if is_age_verified == 'true':
-                self.user_profile['is_age_verified'] = True
+                self.user_profile[config.ATTRIBUTE_IS_AGE_VERIFIED] = True
+                self.profile[config.ATTRIBUTE_IS_AGE_VERIFIED] = Attribute(is_age_verified, True, anchors)
                 return
             if is_age_verified == 'false':
-                self.user_profile['is_age_verified'] = False
+                self.user_profile[config.ATTRIBUTE_IS_AGE_VERIFIED] = False
+                self.profile[config.ATTRIBUTE_IS_AGE_VERIFIED] = Attribute(is_age_verified, False, anchors)
                 return
 
         raise TypeError("age_verified_field unable to be parsed")
 
-    def try_convert_structured_postal_address_to_dict(self, field):
-        decoder = json.JSONDecoder(object_pairs_hook=collections.OrderedDict)
-        self.user_profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS] = decoder.decode(field.value)
-
-    def set_address_to_be_formatted_address_if_null(self):
-        if 'postal_address' not in self.user_profile and 'structured_postal_address' in self.user_profile:
-            if 'formatted_address' in self.user_profile['structured_postal_address']:
-                self.user_profile['postal_address'] = self.user_profile['structured_postal_address'][
-                    'formatted_address']
+    def try_convert_structured_postal_address_to_dict(self, field, anchors):
+        decoder = json.JSONDecoder(object_pairs_hook=collections.OrderedDict, strict=False)
+        value_to_decode = field.value
+        if not isinstance(value_to_decode, str):
+            value_to_decode = value_to_decode.decode()
+
+        self.user_profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS] = decoder.decode(value_to_decode)
+        self.profile[config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS] = Attribute(
+            config.ATTRIBUTE_STRUCTURED_POSTAL_ADDRESS,
+            decoder.decode(value_to_decode),
+            anchors)
+
+    def set_address_to_be_formatted_address_if_null(self, anchors):
```

Comment:
It is certainly better to have descriptive names than comments.  But if the names start to resemble a sentence, it might be going a bit far.  Could this be named something like "set_fallback_address", "ensure_postal_address" or "populate_from_structured_address"?